File: backend/animal-registry/src/main/java/com/animalshelter/animalregistry/dto/CreateAnimalRequest.java
package com.animalshelter.animalregistry.dto;

import com.animalshelter.animalregistry.model.AnimalCategory;
import com.animalshelter.animalregistry.model.Gender;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record CreateAnimalRequest(
        @NotBlank(message = "Name is required")
        @Size(max = 100, message = "Name must be at most 100 characters")
        String name,

        @NotNull(message = "Category is required")
        AnimalCategory category,

        String breed,

        @NotNull(message = "Gender is required")
        Gender gender,

        Integer ageMonths,

        Double weight,

        String color,

        String chipId,

        String description
) {}

File: frontend/src/app/dto/animal/CreateAnimalRequest.ts
import { AnimalCategory } from './AnimalCategory';
import { Gender } from './Gender';

export interface CreateAnimalRequest {
  name: string;
  category: AnimalCategory;
  breed?: string;
  gender: Gender;
  age_months?: number;
  weight?: number;
  color?: string;
  chip_id?: string;
  description?: string;
}
